### PR TITLE
Pixel unit colors based on civilization colors

### PR DESCRIPTION
**Pull request by r3versi**
The idea is to have pixel unit sprites composed of 3 layers (it is not mandatory):
- the **first layer** is the base sprite we are used to: for example for Warrior ("TileSets/FantasyHex/Units/Warrior");
>Warrior
- the **second layer** is a white sprite (can have transparency for _pro_ shading) that will be colored with the nation **inner color** ("TileSets/FantasyHex/Units/Warrior-1");
> Warrior-1
- the **third layer** is just like the second, but it will be colored with the nation **outer color** ("TileSets/FantasyHex/Units/Warrior-2").
> Warrior-2

I implemented a generic method `getLayeredImageColored` that could be useful in the future to support image layers for different kind of entities.

The rules for naming layers are:
* A layer name must start with the same name as the base layer. I.e. `Warrior`, followed by a `-` and any number of digits.
* The layers are ordered alphabetically by their names. 

Here is an example of the effect obtained by using the following two detail layers for warriors:

warriors-shaded

### CODE REVIEW

File: core/src/com/unciv/ui/utils/ImageGetter.kt [hunk 1]
```diff
@@ -67,6 +67,50 @@ object ImageGetter {
...
+     *          - TileSets/FantasyHex/Units/Warrior-1
+     *          - TileSets/FantasyHex/Units/Warrior-2
+     *
+     *      It returns a list in which first layer has unmodified colors, the second is colored in
+     *      GOLD and the third in RED.
+     */
+    fun getLayeredImageColored(baseFileName: String, vararg colors: Color?): ArrayList<Image> {
+        val layerList = arrayListOf<Image>()
+        val layers = textureRegionDrawables.keys
+                .filter { it.startsWith(baseFileName) }
```
Comment (yairm210) [inline core/src/com/unciv/ui/utils/ImageGetter.kt:101]: Not good - some units could start with the name of other units, especially in mods
Example: Fighter, Fighter Jet
Warrior, Warrior prince
Zombie, Zombie legion

Comment (r3versi) [reply core/src/com/unciv/ui/utils/ImageGetter.kt]: I changed the rules for filtering. It is now based on a regex. Here you can find an example https://regex101.com/r/TgintQ/1

**Comment by ravignir** [comment]
What if layers were put in another directory? Would that work?

File: core/src/com/unciv/ui/utils/ImageGetter.kt [hunk 2]
```diff
@@ -67,6 +67,51 @@ object ImageGetter {
...
+     *      "TileSets/FantasyHex/Units/Warrior-NUMBER" are retrieved alphabetically:
+     *          - TileSets/FantasyHex/Units/Warrior
+     *          - TileSets/FantasyHex/Units/Warrior-1
+     *          - TileSets/FantasyHex/Units/Warrior-2983
+     *
+     *      It returns a list in which first layer has unmodified colors, the second is colored in
+     *      GOLD and the third in RED.
+     */
+    fun getLayeredImageColored(baseFileName: String, vararg colors: Color?): ArrayList<Image> {
+        val regex = Regex("^$baseFileName(?>-\\d+)?$")
```
Comment (yairm210) [inline core/src/com/unciv/ui/utils/ImageGetter.kt:99]: Regexes are notoriously heavy lifting. Running a regex on EVERY unit tile (when creating the map, which is every turn) means hundreds of regex checks per tile, times hundreds of tiles.
Another problem is that in the case that we go up to 11 (hard to imagine but let's roll with it), 11 is alphabetically between 1 and 2.

Instead, lets do like
i=1
while(textureRegionDrawables.containsKey("$baseFileName-$i"){
 ...
 i++
}

This DOES mean that if you're missing, say, 2 and you have 3, it won't be rendered. But I think that's fair.

Alternatively, we could have them hard-named as $baseFileName, $baseFileName-innerColor, $baseFileName-outerColor

Comment (r3versi) [reply core/src/com/unciv/ui/utils/ImageGetter.kt:99]: Good points! I believe regex performances relies on patterns complexity, anyway it is for sure much more performant to look for an exact key. I think modders and artists would like to have as much freedom as possible, so I'm going with the loop solution.

**Comment by AdityaMH** [comment]
Hmmm... how about tile improvement color like citadel?

**Pull request merged**